### PR TITLE
SuperPoint TensorRT implementation

### PR DESCRIPTION
**Pull request by xiang-wuu**
Code in this PR implements baseline PyTorch NN using TensorRT Network API, also provide instructions to run demo script by inferring TensorRT generated engine file along with pre-trained PyTorch weight model, the updated code base of original [magicleap/SuperPointPretrainedNetwork](https://github.com/magicleap/SuperPointPretrainedNetwork) is maintained in  [xiang-wuu/SuperPointPretrainedNetwork ](https://github.com/xiang-wuu/SuperPointPretrainedNetwork)

### CODE REVIEW

**Comment by wang-xinyu** [comment]
@xiang-wuu Thanks for the PR!

Can you please share one output image generated from your code?

And does this support tensorrt8.x or 7.x?

**Comment by xiang-wuu** [comment]
@wang-xinyu 

Updated the README for sample output result, it's almost identical for PyTorch and TensorRT implementation.

I have tested it only with tensorrt7.x but it should also work with tensorrt8.x as well!

I haven't done the inference time comparison but i put it as separate task under speed/accuracy validation as future work.

**Comment by wang-xinyu** [comment]
@xiang-wuu Thanks a lot! merging it.

BTW, the 8.x has slightly different APIs from 7.x, please do consider to test with 8.x in the future work if you got any chance.

**Review by wang-xinyu — APPROVED** [review]
